spec/unit/builders/ProjectBuilder.spec.js

```diff
@@ -0,0 +1,289 @@
+/*
+    Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+*/
+
+const fs = require('fs');
+const path = require('path');
+const Q = require('q');
+const rewire = require('rewire');
+
+const CordovaError = require('cordova-common').CordovaError;
+
+describe('ProjectBuilder', () => {
+    const rootDir = '/root';
+
+    let builder;
+    let ProjectBuilder;
+    let spawnSpy;
+
+    beforeEach(() => {
+        spawnSpy = jasmine.createSpy('spawn').and.returnValue(Q.defer().promise);
```

Comment:
Does the calling code expect spawn to return a Q instance?

Reply:
Yes, it uses `.progress`, which is not part of the Promise spec. It is only used as a workaround for [CB-9971](https://issues.apache.org/jira/browse/CB-9971), so it should be possible to remove it at some point.